tractor/_portal.py

```diff
@@ -83,19 +85,37 @@ async def receive(self):
                 "Received internal error at portal?")
             raise unpack_error(msg, self._portal.channel)
 
+    @contextmanager
+    def shield_channel(
+        self
+    ) -> typing.AsyncGenerator['StreamReceiveChannel', None]:
```

Comment:
Lol this type sig is [obv wrong](https://github.com/goodboy/tractor/pull/171/checks?check_run_id=1568199138); originally made it async, for no good reason..